# scRNA-seq info

Notes on scRNA-seq analysis

## Significance

scRNA-seq has emerged as a standard technology to study and enhance the 
understanding of transcription. The space here is dedicated to collect
papers, methods, tutorials, and stuff related to the technology. It has 
been more manually curated and content checked since 2020 April.

## Welcome to single cell world

### Tutorials and scRNA-seq 101

* [Full introductory course of scRNA-seq data analysis from Hemberg lab](https://scrnaseq-course.cog.sanger.ac.uk/website/index.html)
* [Seurat tutorial from Satija lab](https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html)
* [Complete course from Yeo's Lab](https://github.com/YeoLab/single-cell-bioinformatics)
* [Single cell RNA-seq tutorial from Theis' lab](https://github.com/theislab/single-cell-tutorial)
* [Tutorial from Broad Institute](https://broadinstitute.github.io/2019_scWorkshop/)
* [HBC Single-cell RNA-seq: raw sequencing data to counts](https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionIV/lessons/SC_pre-QC.html)

### Review papers

* [Hwang et al, Nature EMM, 2018. Single-cell RNA sequencing technologies and bioinformatics pipelines](https://www.nature.com/articles/s12276-018-0071-8)
* [Vieth et al, Nature Comm, 2019. A systematic evaluation of single cell RNA-seq analysis pipelines](https://www.nature.com/articles/s41467-019-12266-7)
* [Luecken and Theis, Mol Sys Biol, 2019. Current best practices in single‐cell RNA‐seq analysis: a tutorial](https://www.embopress.org/doi/10.15252/msb.20188746) 
* [Cheng et al., Frontier in Genetics, 2019. Single-Cell RNA-Seq Technologies and Related Computational Data Analysis](https://www.frontiersin.org/articles/10.3389/fgene.2019.00317/full)
* [Haque et al., Genome Medicine. A practical guide to single-cell RNA-sequencing for biomedical research and clinical applications](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-017-0467-4)
* [Lafzi et al., Nature Protocols, 2018. Tutorial: guidelines for the experimental design of single-cell RNA sequencing studies](https://www.nature.com/articles/s41596-018-0073-y)
* [Chen et al., Front. Genet., 2019. Single-Cell RNA-Seq Technologies and Related Computational Data Analysis](https://www.frontiersin.org/articles/10.3389/fgene.2019.00317/full)

### Challenges and open questions
* [Lahnemann et al., Genome Biologym 2020. Eleven grand challenges in single-cell data science](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1926-6)


### Experimental design

* [Molin and Camillo, Brif. in Bioinfo., 2019. How to design a single-cell RNA-sequencing experiment: pitfalls, challenges and perspectives](https://academic.oup.com/bib/article/20/4/1384/4831233)
* [Zhang et al., Nat. Comm., 2020. Determining sequencing depth in a single-cell RNA-seq experiment](https://www.nature.com/articles/s41467-020-14482-y)


### Batch effect and technical noise

* [Lun et al., Genome Biology, 2019. EmptyDrops: distinguishing cells from
empty droplets in droplet-based single-cell
RNA sequencing data](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y)


### Imputation

* [Talwar et al., Scientific Reports, 2018. AutoImpute: Autoencoder based imputation of single-cell RNA-seq data](https://www.nature.com/articles/s41598-018-34688-x)
* [Andrews and Hemberg, F1000Research, 2019. False signals induced by single-cell imputation](https://f1000research.com/articles/7-1740)
* [Huang et al., Nat. Methods, 2018. SAVER: gene expression recovery for single-cell RNA sequencing](https://www.nature.com/articles/s41592-018-0033-z)
* [Peng et al., Genome Biology, 2019. SCRABBLE: single-cell RNA-seq imputation constrained by bulk RNA-seq data](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1681-8)


### List of packages and collections

* [Packages collected by Sean Davis](https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html)
* [Collection from CrazyHotTommy](https://github.com/crazyhottommy/RNA-seq-analysis#single-cell-rna-seq)
* [scRNA-tools](https://www.scrna-tools.org/)
* [A collection of 55 trajectory inference methods](https://github.com/dynverse/dynmethods#list-of-included-methods)


### Resources

* [Datasets from 10XGenomics](https://support.10xgenomics.com/single-cell-gene-expression/datasets)
* [Manually curated database from Svensson's lab](http://www.nxn.se/single-cell-studies/)
* [Human and mouse data from Hemberg's lab](https://hemberg-lab.github.io/scRNA.seq.datasets/)


### Comparison with bulk RNA-seq
* [Bacher and Kendziorski, Genome Biology, 2016. Design and computational analysis of
single-cell RNA-sequencing experiments](https://genomebiology.biomedcentral.com/track/pdf/10.1186/s13059-016-0927-y)


